### PR TITLE
Demo

### PR DESCRIPTION
**Pull request by zilionis**

### CODE REVIEW

File: assets/js/issueList.jsx [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+                            />
+                        </div>
+                        <div className="issueFoot">
+                            <Pagination
+                                hideDisabled
+                                activePage={this.state.pagination.currentPage}
+                                itemsCountPerPage={this.state.pagination.per_page}
+                                totalItemsCount={this.state.pagination.last}
+                                pageRangeDisplayed={5}
+                                onChange={this.handlePageChange.bind(this)}
```
[the remark this answers is not in the document]
Comment (zilionis) [reply assets/js/issueList.jsx]: good tip. 👍

File: assets/css/app.scss [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+      font-size: 20px;
+      img {
+        vertical-align: middle;
+      }
+    }
+
+  }
+}
+
+$pagination-circular-color: rgb(159, 213, 51);
```
[the remark this answers is not in the document]
Comment (zilionis) [reply assets/css/app.scss]: 👍

File: assets/css/app.scss [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+
+// ul.pagination>li>a, ul.pagination>li>span {
+//   border-radius: 50% !important;margin: 0 5px;
+//   border: 1px solid $pagination-circular-color;
+// }
```
[the remark this answers is not in the document]
Comment (zilionis) [reply assets/css/app.scss]: agree, somehow forgot this one.

File: assets/js/404.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import React from 'react'
+import Header from './header'
+
+const NoMatch = ({ location }) => (
```
[the remark this answers is not in the document]
Comment (zilionis) [reply assets/js/404.js]: 👍

[the remark this answers is not in the document]
Comment (zilionis) [reply assets/js/404.js]: :1

File: assets/js/app.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+import PageNotFound from './404'
+import Login from './login'
+import IssueList from './issueList'
+
+const App = () => (
+    <Router>
+            <Switch>
+                <Route exact path="/" render={() => (
+                    isAuth ? (
```
[the remark this answers is not in the document]
Comment (zilionis) [reply assets/js/app.js]: As i wrote, I never wrote react code before. And i havn't finished FED, side. From another hand is not sure what api endpoints could be public, which are not. App uses github api endpoints, which not require authorization. You can get api limit errors in some cases.

File: docker/php7-fpm/Dockerfile [hunk 6]
```diff
@@ -0,0 +1 @@
+FROM php:7.2-fpm
```
[the remark this answers is not in the document]
Comment (zilionis) [reply docker/php7-fpm/Dockerfile]: in production images are immutable. you can have dozen images and i imagine you will not build dependencies while creating image. You may use artifacts which are build in CI.

File: src/Controller/IssueController.php [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+
+    public function __construct(Github $github)
+    {
+        $this->githubIssues = $github->api(Github::API_ISSUES);
+    }
+
+    /**
+     * @param string $user
+     * @param string $repo
+     * @Route("/api/{user}/{repo}/{issue}")
```
[the remark this answers is not in the document]
Comment (zilionis) [reply src/Controller/IssueController.php]: yeah I agree with this one.

File: src/Service/HttpClient/HttpClient.php [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+        $this->request= $request->getCurrentRequest();
+        $this->guzzleClient = $client;
+    }
+
+    public function get($uri, $options): ?ResponseInterface
+    {
+        $response = null;
+        try {
+            $accessToken =  $this->request->getSession()->get('accessToken');
+            if (!isset($options['headers']['Authorization']) && $accessToken) {
```
Comment (velnez) [inline src/Service/HttpClient/HttpClient.php]: Try/catch blocks should be not include conditional logic. Move the authorization header related code outside the try/catch, leaving only the guzzle request inside.

File: src/Service/Github/ResponseParser.php [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+    }
+
+    public static function getPagination(ResponseInterface $response): ?array
+    {
+        $headerLink = $response->getHeader('Link');
+        if ($headerLink === null) {
+            return null;
+        }
+
+        $sequence = [
```
Comment (velnez) [inline src/Service/Github/ResponseParser.php]: This data should be moved outside the function. I would suggest a class constant.

**Pull request closed without merging**